plugins/GeoIp2/GeoIP2AutoUpdater.php

```diff
@@ -443,6 +444,37 @@ public static function setUpdaterOptions($options)
         }
     }
 
+    protected static function checkGeoIPUpdateUrl($url)
+    {
+        if (empty($url)) {
+            return;
+        }
+
+        $parsedUrl = @parse_url($url);
+        $schema = $parsedUrl['scheme'] ?? '';
+        $host = $parsedUrl['host'] ?? '';
+
+        if (empty($schema) || empty($host) || !in_array(Common::mb_strtolower($schema), ['http', 'https'])) {
+            throw new Exception(Piwik::translate('GeoIp2_MalFormedUpdateUrl', '<i>'.$url.'</i>'));
+        }
+
+        $validHosts = Config::getInstance()->General['geolocation_download_from_trusted_hosts'];
+        $isValidHost = false;
+
+        foreach ($validHosts as $validHost) {
+            if (substr($host, -strlen($validHost)) === $validHost) {
```

Comment:
Would this also return true for `abcdefgdb-ip.com`?

Reply:
Tested, it does.

Reply:
changed that